src/pip/_internal/utils/misc.py

```diff
@@ -948,6 +948,10 @@ def call_subprocess(
     return ''.join(all_output)
 
 
+def write_output(message):
```

Comment:
I suggest changing this to take *args and pass them on to logger.info.

Reply:
Sounds good. 

With this change i can revert the logging to the %s style for string formatting or keep the change to .format() . I personally prefer the format option but am happy to revert it back.

Reply:
He's saying (and I agree) that given this--

```python
logger.info(
    "%s %s requires %s, which is not installed.",
    project_name, version, dependency[0],
)
```

you should do this--

```python
write_output(
    "%s %s requires %s, which is not installed.",
    project_name, version, dependency[0],
)
```

rather than this--

```python
write_output("{} {} requires {}, which is not installed."
             "".format(project_name, version, dependency[0]))
```

(So only change `logger.info` to `write_output` at the call sites.)

Reply:
Thank you for the clarification on that. I reverted the commands back to the %s method.